pandas/_typing.py

```diff
@@ -122,10 +129,7 @@
 Axes = Collection[Any]
 
 # dtypes
-NpDtype = Union[str, np.dtype]
-Dtype = Union[
-    "ExtensionDtype", NpDtype, type_t[Union[str, float, int, complex, bool, object]]
-]
+Dtype = Union["ExtensionDtype", NpDtype]
```

Comment:
do we need to worry about strings like "categorical" that can be interpreted as pandas dtypes but not np.dtype?

Reply:
I think we are only using `NpDtype` where we are expecting Numpy dtypes